NameGame/NameGame/AppDelegate.swift

```diff
@@ -27,10 +28,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         super.init()
     }
 
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions:
+        [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+        guard let window = self.window, let navigationController = window.rootViewController as? UINavigationController else {
+            preconditionFailure("Expected 'UINavigationController' as window root view controller.")
+        }
+        appCoordinator = AppCoordinator(store: store, container: navigationController)
+        store.subscribe(logger)
         FilePath.createDirectories()
```

Comment:
Let's call `createDirectories` before we set everything else up. Just in case something we set up depends on it.